source/module_base/spherical_bessel_transformer.cpp

```diff
@@ -34,39 +34,39 @@ class SphericalBesselTransformer::Impl
     Impl(Impl const&) = delete;
     Impl& operator=(Impl const&) = delete;
 
-    void radrfft(const int l,
-                 const int ngrid,
-                 const double cutoff,
-                 const double* const in,
-                 double* const out,
-                 const int p = 0,
-                 const bool deriv = false
+    void radrfft(int l,
+                 int ngrid,
+                 double cutoff,
+                 const double* in,
+                 double* out,
+                 int p = 0,
+                 bool deriv = false
     );
 
-    void direct(const int l,
-                const int ngrid_in,
-                const double* const grid_in,
-                const double* const in,
-                const int ngrid_out,
-                const double* const grid_out,
-                double* const out,
-                const int p = 0,
-                const bool deriv = false
+    void direct(int l,
```

Comment:
why delete const?

Reply:
Parameter 'l' is const-qualified in the function declaration; const-qualification of parameters only has an effect in function definitions. See https://clang.llvm.org/extra/clang-tidy/checks/readability/avoid-const-params-in-decls.html

Reply:
Will it be good to be consistent between declaration and definitions? I don't see a reason to make these two codes different

Reply:
> Will it be good to be consistent between declaration and definitions? I don't see a reason to make these two codes different

I see. The point is that declarations should be put in header files; the declarations here are not top-level, hence `const` qualifier has no effect here, making a possible definitions without const possible. @jinzx10 What's your consideration here?

However, in this context, I agree with you that preserve `const` would be better.